### DISCUSSION TITLE
Keycloak 20+ docker image hangs on launch with fips enabled on RHEL Centos 7

### DISCUSSION THREAD

**Discussion by zjesse713** [keycloak/keycloak#16376] [Q&A]
During deployment the only logs I see in the terminal is `Updating the configuration and installing your custom providers, if any. Please wait.`

Dockerfile:

```
  FROM quay.io/keycloak/keycloak:20.0.2 as builder  
  # Enable health and metrics support
  ENV KC_HEALTH_ENABLED=true
  ENV KC_METRICS_ENABLED=true
  
  #Get SPI and save to providers directory
  RUN curl -sL http://myexample.jar
  
  RUN /opt/keycloak/bin/kc.sh build
  
  
  FROM quay.io/keycloak/keycloak:20.0.2
  COPY --from=builder /opt/keycloak/ /opt/keycloak/
  
  WORKDIR /opt/keycloak
    
  # change these values to point to a running postgres instance
  ENTRYPOINT ["/opt/keycloak/bin/kc.sh","start-dev"]
```

I do have fips enabled in the environment
Keycloak version: 20.0.1+

Workaround:
Adding environment variable "JAVA_OPTS_APPEND="-Dcom.redhat.fips=false" resolves the issue but seems to be a temporary fix.

**Comment by gschaden** [comment]
Does your image work on you local machine?
Did that work with a previous version?

**Reply by zjesse713** [reply]
Yes, Image works on local machine it gets passed checking for spi and deploys.
Previous versions that worked in my openshift cluster 10.0.2 and 14.0.0

**Reply by gschaden** [reply]
can you change your startup command 
`/opt/keycloak/bin/kc.sh --verbose start --log-level=DEBUG
`

**Reply by zjesse713** [reply]
After updating the startup command with "/opt/keycloak/bin/kc.sh --verbose start --log-level=DEBUG ". The logs in the terminal still only provides "Updating the configuration and installing your custom providers, if any. Please wait." No further logs.

**Reply by adrian-moisa** [reply]
I have the same problem on a mac m1 with version `quay.io/keycloak/keycloak:22.0.5 start-dev`. Last year the same version worked.

**Reply by Bishop19** [reply]
Same as @adrian-moisa

**Comment by activenode** [comment]
Weirdly I have the exact same issue on a DO K8s cluster and weirdly `redhat.fips=false` helped. How did you come across this ENV change MIGHT help?
